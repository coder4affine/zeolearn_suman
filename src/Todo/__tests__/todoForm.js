import React from "react";
import TodoForm from "../todoForm";

function setup() {
  const props = {
    addTodo: jest.fn(),
    todoText: "",
    changeText: jest.fn()
  };

  const Wrapper = shallow(<TodoForm {...props} />);
  return {
    props,
    Wrapper
  };
}

describe("take snap shot", () => {
  it("snapshot", () => {
    const { Wrapper } = setup();
    expect(Wrapper).toMatchSnapshot();
  });
});

describe("test component exist or not", () => {
  it("description", () => {
    const { Wrapper } = setup();
    expect(Wrapper.find("div").exists()).toBe(true);
    expect(Wrapper.find("div").hasClass("container")).toBe(true);

    expect(Wrapper.find("form").exists()).toBe(true);
    expect(Wrapper.find("form").prop("onSubmit")).toEqual(expect.any(Function));

    expect(Wrapper.find("input").exists()).toBe(true);
    expect(Wrapper.find("input").props()).toEqual({
      type: "text",
      name: "todoText",
      value: "",
      onChange: expect.any(Function),
      required: true
    });

    expect(Wrapper.find("button").exists()).toBe(true);
    expect(Wrapper.find("button").prop("type")).toEqual("submit");
    expect(Wrapper.find("button").text()).toEqual("Add Todo");
  });

  it("simulate onsubmit event", () => {
    const { Wrapper, props } = setup();
    const form = Wrapper.find("form");
    form.simulate("submit");
    form.simulate("submit");
    expect(props.addTodo.mock.calls.length).toEqual(2);
  });

  it("simulate onChange event", () => {
    const { Wrapper, props } = setup();
    const input = Wrapper.find("input");
    input.simulate("change");
    input.simulate("change");
    expect(props.changeText.mock.calls.length).toEqual(2);
  });
});
