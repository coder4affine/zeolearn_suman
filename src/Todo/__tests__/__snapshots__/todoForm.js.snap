// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take snap shot snapshot 1`] = `
<div
  className="container"
>
  <h1>
    Todo App
  </h1>
  <form
    onSubmit={[MockFunction]}
  >
    <input
      name="todoText"
      onChange={[MockFunction]}
      required={true}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</div>
`;
